XLS-0070d-credentials/README.md

```diff
@@ -0,0 +1,573 @@
+|`UnauthorizeCredentials` | |`array`|`STArray`|The credential(s) whose preauthorization should be revoked.|
+
+#### 4.1.1. `AuthorizeCredentials` and `UnauthorizeCredentials`
+
+These fields follow the same rules outlined in section 3.1.3 for the `Credential` object's `AuthorizeCredentials` field. 
+
+### 4.2. Failure Conditions
+* Existing failure conditions for `DepositPreauth` will still be obeyed.
+* None or more than one of `Authorize`, `Unauthorize`, `AuthorizeCredentials`, and `UnauthorizeCredentials` are included (i.e. there must be exactly one of these fields included).
+* The issuer of the credential doesn't exist.
+* If `UnauthorizeCredentials` is included in the transaction, the credential(s) are not currently authorized.
+* If `AuthorizeCredentials` is included in the transaction:
+	* The credential(s) are already authorized.
+	* The account doesn't have enough reserve for the object.
+	* The array is too long (i.e. has more than 8 credentials).
+	* The array is empty (i.e. has no credentials).
+
+### 4.3. State Changes
+
+If the transaction is successful:
+* The `DepositPreauth` object is created or deleted.
+
+## 5. Transaction: `CredentialCreate`
+
+This transaction creates a `Credential` object.
+
+There are two possible methods of doing so:
+* The issuer submits this transaction, and `CredentialAccept` is used for the subject to accept the transaction.
+* The credential data is passed off-chain from the issuer to the subject. The subject then submits the credential themselves, alongside an issuer signature, to the ledger.
+
+### 5.1. Fields
+
+| Field Name | Required? | JSON Type | Internal Type | Description |
+|------------|-----------|-----------|---------------|-------------|
+|`TransactionType`| ✔️|`string`|`UInt16`|The transaction type (`CredentialCreate`).|
+|`Account`| ✔️|`string`|`AccountID`|The subject or issuer of the credential.|
+|`Subject`| |`string`|`AccountID`|The subject of the credential.|
+|`Issuer`| |`string`|`AccountID`|The issuer of the credential.|
+|`CredentialType`|✔️|`string`|`Blob`|A (hex-encoded) value to identify the type of credential from the issuer.|
+|`Signature`| |`string`|`Blob`|A signature from the issuer attesting to the credential.|
+|`Expiration`| |`number`|`UInt32`|Optional credential expiration.|
+|`URI`| |`string`|`Blob`|Optional additional data about the credential (such as a link to the VC document).|
+
+#### 5.1.1. `Subject` and `Issuer`
+
+Only one of these two fields must be specified. If `Subject` is specified, `Account` is the issuer. If `Issuer` is specified, `Account` is the subject.
+
+#### 5.1.2. `Signature`
+
+If the subject is submitting the account, a signature must be included from the issuer attesting to the credential. This signature will sign a simple JSON that looks something like this:
+```typescript
+{
+    Issuer: "rISABEL......",
+    Subject: "rALICE.......",
+    CredentialType: "123ABC",
+}
+```
```

Comment:
Shouldn't the signed data also include the expiration, if one is defined? Otherwise, in the flow where the subject creates the credential using this signature, they could potentially "renew" or extend a credential by deleting it and creating a new one with a different `Expiration` value using the same old signature.

Reply:
That's a good point, added.